### PR TITLE
[docs] Update provider.md for newest provider

### PR DESCRIPTION
**Pull request by jolies93**
Corrected the following that we encountered when using this documentation:
- Added to the callback url path - without this correction we received errors
- Removed the SCOPES - this wouldn't initialize for us, got errors until we removed it

We believe this may have something to do with backstage version or dependency changes - the url hadn't been changed in 3 years and the SCOPES were added 9 months ago

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/atlassian/provider.md [hunk 1]
```diff
@@ -46,18 +46,16 @@ auth:
...
         clientSecret: ${AUTH_ATLASSIAN_CLIENT_SECRET}
-        scope: ${AUTH_ATLASSIAN_SCOPES}
         signIn:
           resolvers:
             # See https://backstage.io/docs/auth/atlassian/provider#resolvers for more resolvers
             - resolver: usernameMatchingUserEntityName
 ```
 
-The Atlassian provider is a structure with three configuration keys:
+The Atlassian provider is a structure with two configuration keys:
```
Comment (awanlin) [inline docs/auth/atlassian/provider.md:55]:
```suggestion
The Atlassian provider is a structure with the following configuration keys:
```

File: docs/auth/atlassian/provider.md [hunk 2]
```diff
@@ -46,18 +46,16 @@ auth:
       development:
```
Comment (awanlin) [inline docs/auth/atlassian/provider.md:46]: Would it make sense to include all the config here? There are some items that are missing, though optional, that would be good to see here. 

https://github.com/backstage/backstage/blob/master/plugins/auth-backend-module-atlassian-provider/config.d.ts

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for the updates @jolies93, left a few small comments.

File: docs/auth/atlassian/provider.md (hunk 2 above)
Comment (jolies93) [reply docs/auth/atlassian/provider.md:46]: @awanlin - added the optional parameters - thanks for the link that made it easy!

**Pull request closed without merging**